conda/plugins/subcommands/doctor/health_checks.py

```diff
@@ -30,6 +35,43 @@ def find_packages_with_missing_files(prefix: str | Path) -> dict[str, list[str]]
     return packages_with_missing_files
 
 
+def find_altered_packages(prefix: str | Path) -> dict[str, list[str]]:
+    """Finds altered packages"""
+    altered_packages = {}
+
+    def generate_sha256_checksum(filepath) -> str:
+        try:
+            with open(filepath, "rb") as f:
+                bytes = f.read()
+                hash = hashlib.sha256(bytes).hexdigest()
+                return hash
+        except OSError as err:
+            raise CondaError(
+                f"Could not generate checksum for file {filepath}"
+                + f" because of the following error: {err}."
+            )
```

Comment:
Please use [`conda.gateways.disk.read.compute_sum`](https://github.com/conda/conda/blob/ce0740ae171b4a2755b8c4f924f830e7cdefedcf/conda/gateways/disk/read.py#L74-L84) instead of implementing an own function for this. It doesn't do the error handling as you're doing here, but you can do that where you call `generate_sha256_checksum`. You'll have to catch `OSError` as well, the raised `PathNotFoundError` is a subclass of `OSError`.